src/main/java/uk/gov/hmcts/ccd/customheaders/CustomHeadersFilter.java

```diff
@@ -0,0 +1,62 @@
+package uk.gov.hmcts.ccd.customheaders;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import org.springframework.util.StringUtils;
+import uk.gov.hmcts.ccd.ApplicationParams;
+
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import java.io.IOException;
+
+@Component
+public class CustomHeadersFilter implements Filter {
+
+    private static final Logger LOG = LoggerFactory.getLogger(CustomHeadersFilter.class);
+
+    private final ApplicationParams applicationParams;
+
+    public CustomHeadersFilter(ApplicationParams applicationParams) {
+        this.applicationParams = applicationParams;
+    }
+
+    @Override
+    public void doFilter(ServletRequest request, ServletResponse response, FilterChain filterChain)
+        throws IOException, ServletException {
+        HttpServletRequest httpRequest = (HttpServletRequest) request;
+        HttpServletResponse httpResponse = (HttpServletResponse) response;
+        if (null != applicationParams
+            && null != applicationParams.getCallbackPassthruHeaderContexts()) {
+            applicationParams.getCallbackPassthruHeaderContexts().stream()
+                .filter(StringUtils::hasLength)
+                .forEach(context -> setContextHeader(context, httpRequest, httpResponse));
+        }
+        filterChain.doFilter(request, response);
```

Comment:
Should this be moved to the start of this method to ensure it always runs at the correct time? (Or does it already?)

Reply:
Done